src/ts/components/settingspanel.ts

```diff
@@ -72,28 +103,132 @@ export class SettingsPanel extends Container<SettingsPanelConfig> {
+    // pass event from root page through for backwards compatibility
+    this.getRootPage().onSettingsStateChanged.subscribe(() => {
       this.onSettingsStateChangedEvent();
+    });
 
-      // Attach marker class to last visible item
-      let lastShownItem = null;
-      for (let component of this.getItems()) {
-        component.getDomElement().removeClass(this.prefixCss(SettingsPanel.CLASS_LAST));
-        if (component.isShown()) {
-          lastShownItem = component;
-        }
-      }
-      if (lastShownItem) {
-        lastShownItem.getDomElement().addClass(this.prefixCss(SettingsPanel.CLASS_LAST));
+    this.updateActivePageClass();
+  }
+
+  private updateActivePageClass(): void {
+    this.getPages().forEach((page: SettingsPanelPage, index) => {
+      if (index === this.activePageIndex) {
+        page.getDomElement().addClass(SettingsPanel.ACTIVE_PAGE_CLASS);
+      } else {
+        page.getDomElement().removeClass(SettingsPanel.ACTIVE_PAGE_CLASS);
       }
+    });
+  }
+
+  setActivePageIndex(index: number): void {
+    const targetPage = this.getPages()[index];
+    if (targetPage !== undefined) {
+      this.animateNavigation(targetPage);
+      this.activePageIndex = index;
+      this.navigationStack.push(targetPage);
+      this.updateActivePageClass();
+      targetPage.onActiveEvent();
+    }
+  }
+
+  setActivePage(page: SettingsPanelPage): void {
+    const index = this.getPages().indexOf(page);
+    this.setActivePageIndex(index);
+  }
+
+  popToRootSettingsPanelPage(): void {
+    this.resetNavigation();
+  }
+
+  popSettingsPanelPage() {
+    // pop one navigation item from stack
+    const currentPage = this.navigationStack.pop(); // remove current page
+    const targetPage = this.navigationStack.slice(-1)[0]; // pick target page without removing it
+
+    if (targetPage !== undefined) {
+      this.setActivePage(targetPage);
+    } else {
+      // fallback to root
+      this.popToRootSettingsPanelPage();
+    }
+    currentPage.onInactiveEvent();
+  }
+
+  private resetNavigation(): void {
+    const currentPage = this.navigationStack.slice(-1)[0];
+    if (currentPage) {
+      currentPage.onInactiveEvent();
+    }
+    this.navigationStack = [];
+    this.activePageIndex = 0;
+    this.animateNavigation(this.getRootPage());
+    this.updateActivePageClass();
+  }
+
+  private animateNavigation(targetPage: SettingsPanelPage) {
+    if (!(this.config as SettingsPanelConfig).pageTransitionAnimation)
+      return;
+    // workaround to enable css transition for elements with auto width / height property
+    // css transition does not work with auto properties by definition so we need to calculate 'real'
+    // width / height values to have a nice looking animation
+
+    const domElement = this.getDomElement();
+    const htmlElement = domElement.get(0);
+    // ensure container has real width / height (for first animation)
+    if (htmlElement.style.width === '' || htmlElement.style.height === '') {
```

Comment:
i'm not entirely sure if this is a problem here, but as far as i remember using ele.style.width/height is discuraged. 

see https://stackoverflow.com/questions/294250/how-do-i-retrieve-an-html-elements-actual-width-and-height for alternatives (`getBoundingClientRect` seems to be the way to go)

Reply:
It is intended in this case.